adapters/coap/src/test/java/org/eclipse/hono/adapter/coap/AbstractHonoResourceTest.java

```diff
@@ -117,7 +117,11 @@ void getPutRequestDeviceAndAuthSucceeds(
     @ParameterizedTest
     @CsvSource(value = {
             "true,/telemetry/OTHER_TENANT/device_1,403",
+            "true,/telemetry/OTHER_TENANT/,404",
```

Comment:
Note that previously 403 was returned in this case (PUT request of an authenticated device without device ID in URI path and with non-matching tenant).
FMPOV, returning 404 fits better here.